### DISCUSSION TITLE
how do I know which version of Kafka is actually being used by certain strimzi version

### DISCUSSION THREAD

**Discussion by cstmgl** [strimzi/strimzi-kafka-operator#4760] [Q&A] [answered]
Again sorry if this is a basic question but I could not understand this.
I'm using (over redhat amq-streams) version 0.20.1 of strimzi.
I was looking at the release notes and I could not find this information. 
https://github.com/strimzi/strimzi-kafka-operator/releases/tag/0.20.1

I'm looking for this information because I'm experiencing a problem that is similar to this: https://issues.apache.org/jira/browse/KAFKA-10387

and I wonder if this might be related to the kafka version actually running on my platform but I could not find which version of kafka is actually used, I could only see the operator version is there any place I can find this information

I'm using image from redhat (this might be the problem) "amq7/amq-streams-kafka-26-rhel7:1.6.2" so I would expect this to be using kafka 2.6 but I did not necesarily understand how redhat and strimzi manage the images
Anyway I was also in the strimzi dockerhub and I could not understood if there was already a strimzi kafka 2.6.1 version or not.

https://hub.docker.com/r/strimzi/kafka/tags?page=1&ordering=last_updated

just for reference this is my config and error:

```
    Transforms:                                         ValueToKey, ExtractField
    Transforms . Extract Field . Field:                 ID
    Transforms . Extract Field . Type:                  org.apache.kafka.connect.transforms.ExtractField$Key
    Transforms . Value To Key . Fields:                 ID
    Transforms . Value To Key . Type:                   org.apache.kafka.connect.transforms.ValueToKey
```

```
Status:
  Conditions:
    Last Transition Time:  2021-04-15T12:16:00.111357Z
    Status:                True
    Type:                  Ready
  Connector Status:
    Connector:
      State:        RUNNING
      Worker _ Id:  10.224.12.219:8083
    Name:           test-jdbc-source
    Tasks:
      Id:     0
      State:  FAILED
      Trace:  org.apache.kafka.connect.errors.ConnectException: org.apache.kafka.common.config.ConfigException: Unknown configuration 'transforms.ValueToKey.type'
              at org.apache.kafka.connect.runtime.ConnectorConfig.transformations(ConnectorConfig.java:296)
              at org.apache.kafka.connect.runtime.Worker.buildWorkerTask(Worker.java:605)
              at org.apache.kafka.connect.runtime.Worker.startTask(Worker.java:555)
              at org.apache.kafka.connect.runtime.distributed.DistributedHerder.startTask(DistributedHerder.java:1251)
              at org.apache.kafka.connect.runtime.distributed.DistributedHerder.access$1700(DistributedHerder.java:127)
              at org.apache.kafka.connect.runtime.distributed.DistributedHerder$10.call(DistributedHerder.java:1266)
              at org.apache.kafka.connect.runtime.distributed.DistributedHerder$10.call(DistributedHerder.java:1262)
```

does strimzi mantain it's own version/fork of kafka itself?

**Comment by tomncooper** [comment, marked as answer]
You seem to be using the stand alone (bare metal) RHEL image there not the Kubernetes image? I would refer to the relevant Red Hat documentation for the OpenShift version of AMQ Streams if you want to run Kafka on Kubernetes. 

However, if you want to run Kafka on bare metal then Strimzi is not the project you need.

**Reply by scholzj** [reply]
@tomncooper There is no standalone RHEL image. That is the image which would be used by AMQ Streams operator.

@cstmgl AMQ Streams is based on Strimzi, but the builds are not identical. So if you use AMQ Streams, it is best to direct your questions to Red Hat support. But I can tell you that the image mentioned above is based on Kafka 2.6.0. So it can be affected by the bug you linked.

**Reply by cstmgl** [reply]
is there a plan for when there will be strimzi support for Kafka 2.6.1

**Reply by scholzj** [reply]
Strimzi supports Kafka 2.6.1.

you can see the supported versions on the website in the Downloads section: https://strimzi.io/downloads/ ... 2.6.1 is supported in 0.22 and 0.22.1.

**Reply by cstmgl** [reply]
yes thank you I found it just now also, 
https://strimzi.io/downloads/ :D guess I need to wait until redhat supports version 0.22.1
